src/index.rs

```diff
@@ -1133,4 +1132,211 @@ mod tests {
         assert!(new2 > old2);
         assert!(new2 > new1);
     }
+
+    const FILE_DIR_1: &str = "folder_1";
+    const FILE_DIR_2: &str = "folder_2";
+    const FILE_NAME: &str = "test_";
+    const FILE_COUNT: i32 = 10;
+
+    fn generate_random_update(
+        root_path: PathBuf,
+        folder_name: Option<&str>,
+        name: Option<&str>,
+    ) -> i32 {
+        let mut rng = rand::thread_rng();
+        let mut rnd_num = rng.gen_range(1..=4);
+
+        let mut folder_1 = root_path.clone();
+        let mut folder_2 = root_path.clone();
+        folder_1.push(FILE_DIR_1);
+        folder_2.push(FILE_DIR_2);
+
+        let mut cur_file_name = "";
+        let mut cur_file_path = folder_1.clone();
+        if let Some(file_name) = name {
+            cur_file_name = file_name.clone();
+            cur_file_path.push(file_name);
+        }
+
+        match rnd_num {
+            // create
+            1 => {
+                let mut file_name_new = String::from(cur_file_name);
+                file_name_new.push_str("_new.txt");
+                create_file_at(
+                    folder_1.clone(),
+                    Some(DATA_SIZE_2),
+                    Some(&file_name_new),
+                );
+            }
+            // update
+            2 => {
+                let mut file = File::create(cur_file_path.as_path())
```

Comment:
The file must exist already, right?

Reply:
> The file must exist already, right?

yes.
But, In tis unit test, the file always exists.